src/constants/analytics.ts

```diff
@@ -1,256 +1,178 @@
+import { RecordOf, UnionOf, ofType, unionize } from 'unionize';
+
 import type { AbacusApiStatus, HumanReadablePlaceOrderPayload } from './abacus';
 import type { OnboardingState, OnboardingSteps } from './account';
-import type { DialogTypes } from './dialogs';
+import { DialogTypesTypes } from './dialogs';
 import type { SupportedLocales } from './localization';
 import type { DydxNetwork } from './networks';
 import { TransferNotificationTypes } from './notifications';
 import type { TradeTypes } from './trade';
 import type { DydxAddress, EvmAddress, WalletConnectionType, WalletType } from './wallets';
 
-type AnalyticsEventDataWithReferrer<T extends AnalyticsEvent> = AnalyticsEventData<T> & {
-  referrer: string;
-};
-export type AnalyticsEventTrackMeta<T extends AnalyticsEvent> = {
-  detail: {
-    eventType: AnalyticsEvent;
-    eventData: AnalyticsEventDataWithReferrer<T>;
-  };
-};
-export type AnalyticsEventIdentifyMeta<T extends AnalyticsUserProperty> = {
-  detail: {
-    property: AnalyticsUserProperty;
-    propertyValue: AnalyticsUserPropertyValue<T>;
-  };
-};
-
 // Do not update. this is used specifically to type how we create custom identify events.
 // If you want to update how identify events work, go to src/lib/analytics.ts
-export const customIdentifyEvent = <T extends AnalyticsUserProperty>(
-  meta: AnalyticsEventIdentifyMeta<T>
-) => {
-  return new CustomEvent('dydx:identify', meta);
+export const customIdentifyEvent = (meta: AnalyticsUserProperty) => {
+  return new CustomEvent('dydx:identify', {
+    detail: { property: meta.type, propertyValue: meta.payload },
+  });
 };
 
 // Do not update. this is used specifically to type how we create custom track events.
 // If you want to update how track events work, go to src/lib/analytics.ts
-export const customTrackEvent = <T extends AnalyticsEvent>(meta: AnalyticsEventTrackMeta<T>) => {
-  return new CustomEvent('dydx:track', meta);
+export const customTrackEvent = (meta: AnalyticsEvent, referrer: string) => {
```

Comment:
would strongly urge against updating this method - the point of this function is to only add typing to the `CustomEvent` object. this fn now has some level of functionality/opinion to it and the nxt time someone wants to add a new property to all `track` events, they'll have to update this function which means people are interacting with the raw `CustomEvent` object again

Reply:
In the version currently in main if you want to add a property to all track events you'd still have to modify this essentially because you'd have to modify `AnalyticsEventTrackMeta` which contains `AnalyticsEventDataWithReferrer`. `customTrackEvent` is colocated and super tied to that thing already. I'm not convinced this is meaningfully less safe in the event of adding more common properties. In the case of adding a new event type it's the same either way.

Reply:
would it? i believe the implementation of the function wouldn't change, only the type `AnalyticsEventDataWithReferrer` would have another property added. the functional changes should all happen in the higher layer functions `track` and `identify`

in the current state this function doesn't hold any value - if you feel really strongly about not reverting then i'd suggest deleting this fn. there's no typesafety provided since a bug introduced into the logic of this code won't be caught by any static typing because `CustomEvent` is untype